spyder/plugins/variableexplorer.py

```diff
@@ -104,15 +104,41 @@ def __init__(self, parent):
     @staticmethod
     def get_settings():
         """
-        Return Variable Explorer settings dictionary
-        (i.e. namespace browser settings according to Spyder's configuration file)
+        Retrieve all Variable Explorer configuration settings
+        
+        Specifically, return the settings in CONF_SECTION with keys in 
+        REMOTE_SETTINGS, and the setting 'dataframe_format'.
+        
+        Returns:
+            dict: settings
         """
         settings = {}
 #        CONF.load_from_ini() # necessary only when called from another process
         for name in REMOTE_SETTINGS:
             settings[name] = CONF.get(VariableExplorer.CONF_SECTION, name)
+
+        # dataframe_format is stored without percent sign in config
+        # to avoid interference with ConfigParser's interpolation
+        name = 'dataframe_format'
+        settings[name] = '%' + CONF.get(VariableExplorer.CONF_SECTION, name)
         return settings
 
+    @Slot(str, object)
+    def change_option(self, option_name, new_value):
+        """
+        Change a config option
+
+        This function is called if sig_option_changed is received. If the
+        option changed is the dataframe format, then the leading '%' character
+        is stripped (because it can't be stored in the user config). Then,
+        the signal is emitted again, so that the new value is saved in the
+        user config.
+        """
+        if option_name == 'dataframe_format':
+            assert new_value.startswith('%')

```

Comment:
That is a good point. I need to think about that.

I think that when I wrote this, I thought it would be a programming error. But now I think this is quite possible if the user writes down a format string with does not begin with a `%`. This may be stupid, but whenever you write a foolproof program a better fool comes along :stuck_out_tongue_winking_eye:  Let me think a bit about this.